File: 226/invert-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {TreeNode}
 */
var invertTree = function (root) {
    if (!root) {
        return null;
    }
    if (!root.left) {
        invertTree(root.left);
    }
    if (!root.right) {
        invertTree(root.right);
    }
    let tmp = root.right;
    root.right = root.left;
    root.left = tmp;
    return root;
};

File: 226/invert-binary-tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func invertTree(root *TreeNode) *TreeNode {
	if nil == root {
		return nil
	}
	if nil != root.Left {
		invertTree(root.Left)
	}
	if nil != root.Right {
		invertTree(root.Right)
	}
	root.Left, root.Right = root.Right, root.Left
	return root
}